File: security-management-frontend/src/services/EntrepriseService.js
// src/services/EntrepriseService.js
import api from "./api";

const EntrepriseService = {
    getAllEntreprises:   ()           => api.get("/entreprises").then(response => response.data),
    getEntrepriseById:   id           => api.get(`/entreprises/${id}`).then(response => response.data),
    createEntreprise:    entreprise   => api.post("/entreprises", entreprise).then(response => response.data),
    updateEntreprise:    (id, ent)    => api.put(`/entreprises/${id}`, ent).then(response => response.data),
    deleteEntreprise:    id           => api.delete(`/entreprises/${id}`).then(response => response.data)
};

export default EntrepriseService;

File: SecurityManagementApp/src/main/java/com/boulevardsecurity/securitymanagementapp/service/EntrepriseService.java
// src/main/java/com/boulevardsecurity/securitymanagementapp/service/EntrepriseService.java
package com.boulevardsecurity.securitymanagementapp.service;

import com.boulevardsecurity.securitymanagementapp.dto.EntrepriseCreateDto;
import com.boulevardsecurity.securitymanagementapp.dto.EntrepriseDto;

import java.util.List;
import java.util.Optional;

public interface EntrepriseService {
    EntrepriseDto createEntreprise(EntrepriseCreateDto dto);
    List<EntrepriseDto> getAllEntreprises();
    Optional<EntrepriseDto> getEntrepriseById(Long id);
    Optional<EntrepriseDto> getEntrepriseByNom(String nom);
    EntrepriseDto updateEntreprise(Long id, EntrepriseDto dto);
    void deleteEntreprise(Long id);
}
